sql-to-dbsp-compiler/lib/sqllib/src/lib.rs

```diff
@@ -580,13 +580,66 @@ pub fn ln_decimal(left: Decimal) -> F64 {
 
 some_polymorphic_function1!(ln, decimal, Decimal, F64);
 
+#[inline(always)]
+pub fn ln_d(left: F64) -> F64 {
+    let left = left.into_inner();
+
+    if left < 0.0 {
+        panic!("Unable to calculate ln for zero")
+    } else if left == 0.0 {
+        panic!("Unable to calculate ln for negative numbers")
+    }
+
+    left.ln().into()
+}
+
+some_polymorphic_function1!(ln, d, F64, F64);
```

Comment:
it would be polymorphic if it works for f32 too. Can you check whether we need it?
Otherwise you can probably use some_function!.